x-pack/plugins/data_visualizer/public/application/data_comparison/data_comparison_page.tsx

```diff
@@ -0,0 +1,395 @@
+});
+
+export const PageHeader: FC = () => {
+  const [, setGlobalState] = useUrlState('_g');
+  const { dataView } = useDataSource();
+
+  const [frozenDataPreference, setFrozenDataPreference] = useStorage<
+    DVKey,
+    DVStorageMapped<typeof DV_FROZEN_TIER_PREFERENCE>
+  >(
+    DV_FROZEN_TIER_PREFERENCE,
+    // By default we will exclude frozen data tier
+    FROZEN_TIER_PREFERENCE.EXCLUDE
+  );
+
+  const timefilter = useTimefilter({
+    timeRangeSelector: dataView.timeFieldName !== undefined,
+    autoRefreshSelector: true,
+  });
+
+  const updateTimeState: FullTimeRangeSelectorProps['callback'] = useCallback(
+    (update) => {
+      setGlobalState({
+        time: {
+          from: moment(update.start.epoch).toISOString(),
+          to: moment(update.end.epoch).toISOString(),
+        },
+      });
+    },
+    [setGlobalState]
+  );
+
+  const hasValidTimeField = useMemo(
+    () => dataView.timeFieldName !== undefined && dataView.timeFieldName !== '',
+    [dataView.timeFieldName]
+  );
+
+  return (
+    <EuiPageHeader
+      pageTitle={<div css={dataViewTitleHeader}>{dataView.getName()}</div>}
+      rightSideItems={[
+        <EuiFlexGroup gutterSize="s" data-test-subj="dataComparisonTimeRangeSelectorSection">
+          {hasValidTimeField ? (
+            <EuiFlexItem grow={false}>
+              <FullTimeRangeSelector
+                frozenDataPreference={frozenDataPreference}
+                setFrozenDataPreference={setFrozenDataPreference}
+                dataView={dataView}
+                query={undefined}
+                disabled={false}
+                timefilter={timefilter}
+                callback={updateTimeState}
+              />
+            </EuiFlexItem>
+          ) : null}
+          <DatePickerWrapper
+            isAutoRefreshOnly={!hasValidTimeField}
+            showRefresh={!hasValidTimeField}
+            width="full"
+            flexGroup={false}
+          />
+        </EuiFlexGroup>,
+      ]}
+    />
+  );
+};
+
+export const DataComparisonPage: FC = () => {
+  const {
+    services: { data: dataService },
+  } = useDataVisualizerKibana();
+  const { dataView, savedSearch } = useDataSource();
+
+  const [dataComparisonListState, setAiopsListState] = usePageUrlState<{
+    pageKey: 'DV_DATA_COMP';
+    pageUrlState: DataComparisonFullAppState;
+  }>('DV_DATA_COMP', getDefaultDataComparisonState());
+
+  const [randomSamplerMode, setRandomSamplerMode] = useStorage<
+    DVKey,
+    DVStorageMapped<typeof DV_RANDOM_SAMPLER_PREFERENCE>
+  >(DV_RANDOM_SAMPLER_PREFERENCE, RANDOM_SAMPLER_OPTION.ON_AUTOMATIC);
+
+  const [randomSamplerProbability, setRandomSamplerProbability] = useStorage<
+    DVKey,
+    DVStorageMapped<typeof DV_RANDOM_SAMPLER_P_VALUE>
+  >(DV_RANDOM_SAMPLER_P_VALUE, MIN_SAMPLER_PROBABILITY);
+  const [lastRefresh, setLastRefresh] = useState(0);
+
+  const forceRefresh = useCallback(() => setLastRefresh(Date.now()), [setLastRefresh]);
+
+  const randomSampler = useMemo(
```

Comment:
Have you run into this error when using the random sampler. Here the error occurs even at 50% sampling rate. Is there anything we can do to prevent this, or add a suggestion to the error callout to try increasing / turning off random sampling:

<img width="1012" alt="image" src="https://github.com/elastic/kibana/assets/7405507/877148cc-c081-4791-8fcf-779f4a039be4">

Reply:
Should be fixed with this change [`05245ea` (#161365)](https://github.com/elastic/kibana/pull/161365/commits/05245ea3772158e86e85734432d264dc6708429a)